### ISSUE TITLE
Quarks could make do with an older version of git

### ISSUE THREAD

**Issue by danielmkarlsson** [supercollider/supercollider#5807]
## Environment

* SuperCollider version: 3.12.2
* Operating system: Mac os 10.14.6

## Steps to reproduce

If you are on a clean install, the quarks system asks you to download xcode because it needs a certain version of git. That makes anyone unfamiliar with the process think "Oh no, this is going to be that huge xcode download I've heard about", while it is in fact a somewhat smaller subset which is called xcode select, the prompt on screen for mac users does not make this clear. 

```supercollider
Quarks.install("https://github.com/madskjeldgaard/PolyBuf.git");
```

## What I want

I meet a lot of students with older macs. I would like to discuss here if it were at all possible to make do with the version of git that ships with say, Mac os 10.14.x, for the purposes of using git as part of the quarks system. As not everyone has the possibility or indeed, the desire to buy new computers, I feel that this is an opportunity to do something nice for new SuperCollider users with old computers.

**Comment by dyfer** [comment]
I'm pretty sure I've used SC on a machine with macOS 10.14 using the git from the xcode command line tools. I don't believe there's any problem with that version, but I don't have access to such machine to check right now.

I looked briefly and it looks like our help files don't really talk about the git dependency at all (oops...) - unless I missed something. I think the first steps would be to document Git requirement, either as a separate guide, linked from [Using Quarks](https://doc.sccode.org/Guides/UsingQuarks.html), or added to that guide (I think adding it only to the documentation of [Git](https://doc.sccode.org/Classes/Git.html) would be hard to discover for the user). There we could clarify that on macOS the easiest way to get git is by running `xcode-select install`, and also providing the homebrew alternative. We should also add applicable information for other platforms.

Unfortunately SC's git interface breaks easily (e.g. when there are issues with PATH) so I'm not sure if we could really automate this process well to account for various scenarios...

**Comment by danielmkarlsson** [comment]
I am sorry I was so unclear in my original post. Let me rephrase.
I think it would be beneficial to lower the git version number requirement in the quarks system.
This could let more users use the quarks system more easily, because they would be using the version of git that is already installed on their computer.

**Comment by dyfer** [comment]
Thanks @danielmkarlsson 
Yes, running `git --version` would be a first step. 
If that reports a valid version, then I'd check if cloning works.
In a temporary directory, or on the Desktop, run:
`git clone https://github.com/supercollider-quarks/MathLib`
and report back whether MathLib was cloned successfully.
Also, please report where's the git binary just to double check:
`which git`

**Comment by jamshark70** [comment]
Here is how we issue git commands:

```
	git { |args, cd=true|
		var cmd, result="";

		if(cd, {
			cmd = ["cd", thisProcess.platform.formatPathForCmdLine(localPath), "&&", "git"];
		},{
			cmd = ["git"];
		});
		cmd = (cmd ++ args).join(" ");
		// this blocks the app thread
		Pipe.callSync(cmd, { |res|
			result = res;
		}, {
			Git.checkForGit();
		});
		^result;
	}
```

There's no version enforcement.

We issue a simple "git" command (i.e. there's nothing we can do to make the command simpler), and macOS pops up that dialog.

Perhaps, on mac, we could verify that the executable is real git instead of fake git and show our own message instead...?

**Comment by danielmkarlsson** [comment]
I see. 
Yes, I think that if SC could take charge there, and show a message that indicated that this pertains to xcode select, which is in the megabyte range and not xcode, which is in the tens of gigabytes range, then that would be an improvement.

**Comment by danielmkarlsson** [comment]
I did a clean slate install of macOS 10.14.6 and I am confused by my findings.

`which git` yields `usr\bin\git`

`git --version` yields the prompt that installs xcode select.

I pressed cancel and went snooping around in usr\bin and git is indeed there along with a bunch of git related files.
Not sure what to do now.
I am suspecting I was wrong about there being a git that could be used for the quarks system on a clean install.
I feel silly.

**Comment by dyfer** [comment]
Oh no need need to feel silly. 

> `git --version` yields the prompt that installs xcode select.

That confirms that indeed there's no working git on a clean install of macOS.

At the very least, we should add documentation about git that is not developer-specific. My feeling is that in that case we should suggest installing command line tools instead of homebrew on macOS (seems less intrusive), unless the user already has a homebrew installed.

**Comment by danielmkarlsson** [comment]
I agree that maybe not everyone has homebrew or is comfortable even just opening a terminal so maybe make choices accordingly.
I installed xcode select now on macOS 10.14.6 and it is 2.17GB, which I think is pretty big.
So I uninstalled xcode select thinking that I could get git installed through homebrew but then when I installed homebrew it installed xcode select for me again as part of homebrew's dependencies anyway.
Thank you for your time and sorry to bother everyone.
Should I close this issue?

**Comment by dyfer** [comment]
> I installed xcode select now on macOS 10.14.6 and it is 2.17GB, which I think is pretty big.

ugh, this is quite big.

Maybe this issue could be closed, but I think it would be worthwhile to have a new issue that at least requests the documentation of the git requirement, which is currently lacking, AFAICT.

**Comment by danielmkarlsson** [comment]
Thank you for your patience and time everyone.
